lib/ts/recipe/totp/index.ts

```diff
@@ -0,0 +1,188 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getNormalisedUserContext } from "../../utils";
+import { RecipeFunctionOptions } from "../recipeModule/types";
+import Recipe from "./recipe";
+import {
+    PreAndPostAPIHookAction,
+    PreAPIHookContext,
+    PostAPIHookContext,
+    RecipeInterface,
+    UserInput,
+    DeviceInfo,
+} from "./types";
+
+export default class RecipeWrapper {
+    static init(config?: UserInput) {
+        return Recipe.init(config);
+    }
+
+    /**
+     * Creates a new unverified device.
+     *
+     * @param deviceName (OPTIONAL) A display name for the device
+     *
+     * @param userContext (OPTIONAL) Refer to {@link https://supertokens.com/docs/emailpassword/advanced-customizations/user-context the documentation}
+     *
+     * @param options (OPTIONAL) Use this to configure additional properties (for example pre api hooks)
+     *
+     * @returns `{ status: "OK", ...}` if successful along a description of the created device (secret, etc.)
+     */
+    static createDevice(input: { deviceName?: string; options?: RecipeFunctionOptions; userContext: any }): Promise<
```

Comment:
userContext should be optional, and therefore `input` should also be optional. Please double check this everywhere.